test/integration/api_v1_test.rb

```diff
@@ -586,6 +586,27 @@ class ApiV1Test < ActionDispatch::IntegrationTest
     assert_response :unauthorized
   end
 
+  test 'add a key to a project' do
```

Comment:
Would you mind adding tests for (at least) the forbidden case as well? It would also be nice to have a test for the unprocessable variety - maybe if you name the params wrong that will trigger?